extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/runtime/JpaOperations.java

```diff
@@ -267,6 +268,13 @@ public static Object findById(Class<?> entityClass, Object id, LockModeType lock
         bindParameters(jpaQuery, params);
         return new PanacheQueryImpl(em, jpaQuery, findQuery, params);
     }
+    
+    @SuppressWarnings("rawtypes")
+    public static PanacheQuery<?> findAll(Class<?> entityClass, CriteriaQuery<?> criteriaQuery) {
```

Comment:
I'm not sure if JPA objects should be part of the API, do you have an use case?

Reply:
JPA was not used

Reply:
`CriteriaQuery` is part of the JPA API no?

Reply:
`javax.persistence.criteria. CriteriaQuery`

Reply:
This project already uses: `javax.persistence.*`